lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -1713,9 +1713,12 @@ bool SROAGlobalAndAllocas(HLModule &HLM, bool bHasDbgInfo) {
         a1ty->isStructTy() && a1ty->getStructNumElements() == 1;
     auto sz0 = DL.getTypeAllocSize(a0ty);
     auto sz1 = DL.getTypeAllocSize(a1ty);
-    if (sz0 == sz1 && (isUnitSzStruct0 || isUnitSzStruct1))
-      return getNestedLevelInStruct(a0ty) < getNestedLevelInStruct(a1ty);
-    return sz0 < sz1;
+    if (sz0 == sz1 && (isUnitSzStruct0 || isUnitSzStruct1)) {
+      sz0 = getNestedLevelInStruct(a0ty);
+      sz1 = getNestedLevelInStruct(a1ty);
+    }
+    // If sizes are equal, tiebreak with alphabetically lesser at higher priority
+    return sz0 < sz1 || (bHasDbgInfo && sz0 == sz1 && a0->getName() > a1->getName());
```

Comment:
Are a0 and a1 always guaranteed to have a name?

Reply:
That method is guaranteed to return a StringRef. If there is no name, it returns an empty string. In debugging case, it's at least very likely, but I didn't think it was worth prechecking the presence of a name since there is this acceptable fallback.

Reply:
Don't we have some weirdness where Release build of the compiler vs Debug build of the compiler have different behaviours with giving values names? Not sure if it's limited to certain types of values or if it affects allocas.